pkg/resource/open_id_connect_provider/hooks.go

```diff
@@ -0,0 +1,301 @@
+			latestClientIDs[*latestClientID] = true
+		}
+
+		for desiredClientID, _ := range desiredClientIDs {
+			_, hasLatest := latestClientIDs[desiredClientID]
+			if !hasLatest {
+				// clientID is to be added
+
+				addClientIDInput, err := rm.newAddClientIDRequestPayload(ctx, desired, &desiredClientID)
+				if err != nil {
+					return nil, err
+				}
+
+				var addClientIDResp *svcsdk.AddClientIDToOpenIDConnectProviderOutput
+				_ = addClientIDResp
+				addClientIDResp, err = rm.sdkapi.AddClientIDToOpenIDConnectProviderWithContext(ctx, addClientIDInput)
+				rm.metrics.RecordAPICall("UPDATE", "AddClientIDToOpenIDConnectProvider", err)
+				if err != nil {
+					return nil, err
+				}
+			} else {
+				delete(desiredClientIDs, desiredClientID)
+				delete(latestClientIDs, desiredClientID)
+			}
+		}
+		for latestClientID, _ := range latestClientIDs {
+			// clientID is to be removed
+			removeClientIDInput, err := rm.newRemoveClientIDRequestPayload(ctx, desired, &latestClientID)
+			if err != nil {
+				return nil, err
+			}
+
+			var removeClientIDResp *svcsdk.RemoveClientIDFromOpenIDConnectProviderOutput
+			_ = removeClientIDResp
+			removeClientIDResp, err = rm.sdkapi.RemoveClientIDFromOpenIDConnectProviderWithContext(ctx, removeClientIDInput)
+			rm.metrics.RecordAPICall("UPDATE", "removeClientIDFromOpenIDConnectProvider", err)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	// Merge in the information we read from the API call above to the copy of
+	// the original Kubernetes object we passed to the function
+	ko := desired.ko.DeepCopy()
+
+	rm.setStatusDefaults(ko)
+	if delta.DifferentAt("Spec.Tags") {
+		if err := rm.syncTags(ctx, &resource{ko}); err != nil {
+			return nil, err
+		}
+	}
+	// There really isn't a status of a role... it either exists or doesn't. If
+	// we get here, that means the update was successful and the desired state
+	// of the role matches what we provided...
+	ackcondition.SetSynced(&resource{ko}, corev1.ConditionTrue, nil, nil)
+
+	return &resource{ko}, nil
+}
+
+// compareTags is a custom comparison function for comparing lists of Tag
+// structs where the order of the structs in the list is not important.
+func compareTags(
+	delta *ackcompare.Delta,
+	a *resource,
+	b *resource,
+) {
+	if len(a.ko.Spec.Tags) != len(b.ko.Spec.Tags) {
+		delta.Add("Spec.Tags", a.ko.Spec.Tags, b.ko.Spec.Tags)
+	} else if len(a.ko.Spec.Tags) > 0 {
+		if !commonutil.EqualTags(a.ko.Spec.Tags, b.ko.Spec.Tags) {
+			delta.Add("Spec.Tags", a.ko.Spec.Tags, b.ko.Spec.Tags)
+		}
+	}
+}
+
+// syncTags examines the Tags in the supplied OpenIDConnectProvider and calls the ListOpenIDConnectProviderTags,
+// TagOpenIDConnectProvider and UntagOpenIDConnectProvider API endpoints to ensure that the set of associated Tags stays
+// in sync with the OpenIDConnectProvider.Spec.Tags
+func (rm *resourceManager) syncTags(
+	ctx context.Context,
+	r *resource,
+) (err error) {
```

Comment:
Ideally we'd use the new common runtime tagging constructs and helper methods, but for now this matches the other resources.